### PR TITLE
Move to Pircbotx

### PR DESCRIPTION
**Pull request by slide**
Pircbotx is a fork of Pircbot that supports additional features such as SASL (used for identifiying with services during connection). This will make it easier for the bot to join +r channels.

### CODE REVIEW

**Comment by slide** [comment]
@rtyler, @oleg-nenashev, @daniel-beck, @olblak Please review these changes. I tested the SASL support yesterday and it works like a charm. This would remove any need to delay in the start up to wait for the `/nickserv` to complete before joining channels. The authentication with nickserv happens during connection to the server.

File: src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java [hunk 1]
```diff
@@ -466,79 +484,56 @@ private void replyBugStatus(String channel, String ticket) {
...
-                e.printStackTrace();
-                try {
-                    Thread.sleep(15000);
-                } catch (InterruptedException _) {
-                    return; // abort
-                }
-            }
-        }
+    private boolean isSenderAuthorized(Channel channel, User sender, boolean acceptVoice) {
+        return sender.getUserLevels(channel).stream().anyMatch(e -> e == UserLevel.OP || (acceptVoice && e == UserLevel.VOICE)
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java]: Not the best part of Java 8 IMHO. LINQ--

Comment (slide) [reply src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java]: You hurt my feelings as a .NET Dev ;-)

**Comment by rtyler** [comment]
I have no strong opinions here, let's ship it

**Pull request merged**